framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenes.java

```diff
@@ -223,13 +229,6 @@ public void updateInferredMethodParameterTypes(
             // https://github.com/typetools/checker-framework/issues/682
             return;
         }
-        // TODO: We must handle cases where the method is declared on a superclass.
-        // Currently we are ignoring them. See ElementUtils#getSuperTypes.
-        // See Issue 682
-        // https://github.com/typetools/checker-framework/issues/682
```

Comment:
Does this PR fix this issue?

Reply:
Yes. The test case I added fails without the changes I made.

Reply:
Then you should include `Fixes #682` in the pull request title.

Reply:
Sorry, I misunderstood your question. This PR fixes *the part of 682 described in the code comment I removed*. #682 is a large issue detailing all the known problems with WPI. This PR doesn't fix all of those problems, so closing the issue would be inappropriate.

Reply:
When this is merged, I can update #682 so that it no longer mentions this problem.